glean-core/android/src/main/java/mozilla/telemetry/glean/utils/JSONArray.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.telemetry.glean.utils
+
+import org.json.JSONArray
+import org.json.JSONException
+
+/**
+ * Convenience method to convert a JSONArray into a sequence.
+ *
+ * @param getter callback to get the value for an index in the array.
+ */
+inline fun <V> JSONArray.asSequence(crossinline getter: JSONArray.(Int) -> V): Sequence<V> {
+    val indexRange = 0 until length()
+    return indexRange.asSequence().map { i -> getter(i) }
+}
+
+fun JSONArray.asSequence(): Sequence<Any> = asSequence { i -> get(i) }
+
+/**
+ * Convenience method to convert a JSONArray into a List
+ *
+ * @return list with the JSONArray values, or an empty list if the JSONArray was null
+ */
+@Suppress("UNCHECKED_CAST")
+fun <T> JSONArray?.toList(): List<T> {
```

Comment:
Let's also import the relevant test coverage for the imported functions coming from [this file](https://github.com/mozilla-mobile/android-components/blob/cd3eb4d3473db5f3fd6524decf1c8cee80cc7548/components/support/ktx/src/test/java/mozilla/components/support/ktx/android/org/json/JSONArrayTest.kt)